### ISSUE TITLE
can you get the result from this paper?

### ISSUE THREAD

**Issue by Ttt749** [FishSmile-syx/ITAE-Pytorch-Anomaly_Detection#1]
I train your code, but my result isn't good. can you already get the result of this paper?

**Comment by SMSD75** [comment]
I have this problem too. I think the results are not reproducible.

**Comment by Ttt749** [comment]
Yes, I try to reimplement this code, but I can't get the result, I also think the results are not reproducible. Thanks.

**Comment by chaoqinhuang** [comment]
Thanks for your interest in our paper.
I have no time to check the  code carefully, but I find many questions (just for mnist):
(1), the test set should have 10000 images but not 60000+ images;
(2), remove transforms.Normalize;
(3), use optimizer.zero_grad() correctly;
(4), your dataloader do not have any data augmentation, please add it according to our original paper;
(5), I don't know whether your code for calculating nrm_err is correct. I just remove it for the debug.
(6), (optimal) please replace your L1 measurement to nn.L1Loss(). I don't have enough time to check your code carefully, thus I don't know if there is any bug for your code. I suggest that you should use more official code in Pytorch.
(7), (optimal) please use the roc_auc_score function in sklearn.metrics
(8), (optimal) please use the dataloader in Pytorch, but not write by yourself.

After so many modifications to your codes, the loss value after training 50 epochs is ~0.004 with auroc 99+ for MNIST class 6, but the loss for your original code is still 0.3+ with 60+ auroc.

ITAE experiment on mnist dataset
set 6 to be normal class and other classes to be abnormal class


--------------------------------------------------------------------------------
 2020-08-06 16:31:48	Epoch: 1 	Loss: 0.0443	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:31:57	Epoch: 2 	Loss: 0.0260	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:32:04	Epoch: 3 	Loss: 0.0209	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:32:11	Epoch: 4 	Loss: 0.0175	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:32:17	Epoch: 5 	Loss: 0.0156	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:32:24	Epoch: 6 	Loss: 0.0139	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:32:31	Epoch: 7 	Loss: 0.0127	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:32:38	Epoch: 8 	Loss: 0.0119	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:32:45	Epoch: 9 	Loss: 0.0112	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:32:52	Epoch: 10 	Loss: 0.0105	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:32:59	Epoch: 11 	Loss: 0.0100	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:33:06	Epoch: 12 	Loss: 0.0095	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:33:13	Epoch: 13 	Loss: 0.0090	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:33:20	Epoch: 14 	Loss: 0.0088	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:33:27	Epoch: 15 	Loss: 0.0084	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:33:36	Epoch: 16 	Loss: 0.0081	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:33:42	Epoch: 17 	Loss: 0.0078	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:33:49	Epoch: 18 	Loss: 0.0075	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:33:56	Epoch: 19 	Loss: 0.0075	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:34:04	Epoch: 20 	Loss: 0.0073	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:34:12	Epoch: 21 	Loss: 0.0070	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:34:19	Epoch: 22 	Loss: 0.0069	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:34:27	Epoch: 23 	Loss: 0.0068	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:34:34	Epoch: 24 	Loss: 0.0066	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:34:41	Epoch: 25 	Loss: 0.0064	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:34:48	Epoch: 26 	Loss: 0.0063	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:34:55	Epoch: 27 	Loss: 0.0062	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:35:02	Epoch: 28 	Loss: 0.0060	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:35:10	Epoch: 29 	Loss: 0.0060	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:35:17	Epoch: 30 	Loss: 0.0058	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:35:24	Epoch: 31 	Loss: 0.0058	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:35:32	Epoch: 32 	Loss: 0.0058	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:35:39	Epoch: 33 	Loss: 0.0056	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:35:46	Epoch: 34 	Loss: 0.0055	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:35:53	Epoch: 35 	Loss: 0.0055	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:36:02	Epoch: 36 	Loss: 0.0053	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:36:09	Epoch: 37 	Loss: 0.0052	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:36:16	Epoch: 38 	Loss: 0.0052	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:36:23	Epoch: 39 	Loss: 0.0052	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:36:31	Epoch: 40 	Loss: 0.0051	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:36:38	Epoch: 41 	Loss: 0.0050	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:36:46	Epoch: 42 	Loss: 0.0050	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:36:56	Epoch: 43 	Loss: 0.0050	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:37:03	Epoch: 44 	Loss: 0.0047	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:37:12	Epoch: 45 	Loss: 0.0048	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:37:19	Epoch: 46 	Loss: 0.0048	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:37:26	Epoch: 47 	Loss: 0.0047	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:37:33	Epoch: 48 	Loss: 0.0046	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:37:40	Epoch: 49 	Loss: 0.0046	LR: 0.100000
--------------------------------------------------------------------------------
 2020-08-06 16:37:46	Epoch: 50 	Loss: 0.0045	LR: 0.050000
--------------------------------------------------------------------------------

**Comment by SMSD75** [comment]
the problem is solved?

**Comment by chaoqinhuang** [comment]
This is not our code but just reproduced by others. This reproduced version has so many bugs. But these bugs can be solved. You can try by yourself.
Our paper is under review. We will release our code after be accepted.

**Comment by hyunjunju** [comment]
@chaoqinhuang Is the U-Net in this code is correct implementation?

**Comment by adaxigua** [comment]
> This is not our code but just reproduced by others. This reproduced version has so many bugs. But these bugs can be solved. You can try by yourself. Our paper is under review. We will release our code after be accepted.

Congratulations on your acception of the paper! When would you plan to release your code? I am looking forward to learn it!
